SubZero/src/model/interface/CameraInterface.h

```diff
@@ -9,21 +9,82 @@
 #define CAMERAINTERFACE_H_
 
 #include "HwInterface.h"
+#include "Data.h"
+// include other camera + image specific classes
+// #include "ImageInputProcessor.h"
+// #include "ImageFormatter.h"
+// #inlcude "CamIntrinsic.h" //stores camera intrinsic parameters
+
+enum CameraPosition {
+  FRONT,
+  DOWN
+};
+
+
 /**
  * A child of HwInterface that deals specifically with the camera. Concrete class.
  */
 
 class CameraInterface: public HwInterface {
+
+private:
+
+	/*
+     * pointer to memory allocated for storing raw (but 
+     * decoded) data from hardware
+     */
+    SwData *decodedBuffer[]; 
+
+    /* allocate only enough memory to keep a max of 
+     * this number of hardware inputs; discard all after that
+     */
+    int bufferSize; 
+
+    /* 
+     * look at only every this number of inputs
+     * i.e. pull data from hardware at this frequency
+     */
+    int pullFrequency;
+
+    /* 
+     * this value specifies the encoding, decoding policy to use
+     * alternatively, we can just implement encode(), decode() ... etc
+     * differently for each hardwares when we branch out 
+     * from our abstract class
+     */
+    int policy;
+
+    /* 
+     * id of hardware this interface interacts with
+     * hardware can be accessed (somehow) via this ID
+     */
+    int hardwareID;
+
 public:
-	CameraInterface();
-	/**
-	 * Constructor
-	 */
-	virtual ~CameraInterface();
-	/**
-	 * Deconstructor
-	 */
+
+    /* 
+     * enocde data to be sent to hardware
+     * encode the msg at pointer sData
+     * output results to location at pointer hData
+     */
+    virtual int encode(SwData *sData, HwData *hData); 
```

Comment:
Apparently they will all be collected under DataDefinition.h
Here as Data.h. Will change it.